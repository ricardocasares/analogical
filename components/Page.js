// external
import Head from "next/head";

// internal
import Global from "./Global";
import Header from "./Header";
import Footer from "./Footer";
import Progress from "./Progress";

export default ({ children, title, theme }) => (
  <div className="app">
    {/* Global styles */}
    <Global />
    {/* Inject tags into <head> */}
    <Head>
      <title>{title}</title>
      <meta charSet="utf-8" />
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
    </Head>
    {/* Header and navigation */}
    <Progress />
    {/* Header and navigation */}
    <Header />
    {/* Main content */}
    <main className="page">{children}</main>
    {/* Footer */}
    <Footer />
  </div>
);
