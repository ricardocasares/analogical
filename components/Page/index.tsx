import { FunctionComponent } from "react";
import Box from "@app/components/Box";
import Footer from "@app/components/Footer";
import Navigation from "@app/components/Navigation";

export const Page: FunctionComponent = ({ children }) => (
  <Box
    p={[3, 5]}
    display="flex"
    flexDirection="column"
    minHeight="100%"
    bg="dark.0"
  >
    <Box as="header">
      <Navigation />
    </Box>
    {children}
    <Footer />
  </Box>
);

export default Page;
