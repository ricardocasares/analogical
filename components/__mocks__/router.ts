import Router from "next/router";

// @ts-ignore
Router.router = {
  push: () => {},
  prefetch: () => {}
};
