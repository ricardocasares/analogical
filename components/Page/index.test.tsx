import React from "react";
import renderer from "react-test-renderer";
import serializer from "jest-emotion";
import Page from "@app/components/Page";
import "../__mocks__/router";

expect.addSnapshotSerializer(serializer);

test("renders with correct styles", () => {
  expect(
    renderer
      .create(
        <Page>
          <h1>Hello</h1>
        </Page>
      )
      .toJSON()
  ).toMatchSnapshot();
});
