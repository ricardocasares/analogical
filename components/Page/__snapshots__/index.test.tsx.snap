// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
.emotion-10 {
  background-color: dark.0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100%;
  padding: 16px;
}

@media screen and (min-width:40em) {
  .emotion-10 {
    padding: 64px;
  }
}

.emotion-4 {
  overflow-x: scroll;
  overflow-y: hidden;
  white-space: nowrap;
  -webkit-overflow-scrolling: touch;
}

.emotion-4::-webkit-scrollbar {
  display: none;
}

.emotion-2 {
  margin: 0;
  padding: 0;
  list-style: none;
}

.emotion-2 li {
  display: inline-block;
  margin-right: 10px;
}

.emotion-2 li:last-child {
  margin-right: 0;
}

.emotion-2 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: lowercase;
  -webkit-transition: color 1s;
  transition: color 1s;
  font-size: 16px;
}

.emotion-1 {
  font-weight: 500;
  text-transform: lowercase;
  color: currentcolor;
}

.emotion-0 {
  display: none;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    display: inline;
  }
}

.emotion-9 {
  margin-top: 40px;
}

<div
  className="emotion-10"
  display="flex"
>
  <header
    className="emotion-5"
  >
    <div
      className="emotion-4"
    >
      <nav>
        <ul
          className="emotion-2 emotion-3"
        >
          <li>
            <a
              className={null}
              href="/"
              onClick={[Function]}
            >
              <span
                className="emotion-1"
              >
                ùå∫ 
                <div
                  className="emotion-0"
                  display={
                    Array [
                      "none",
                      "inline",
                    ]
                  }
                >
                  analogic.al
                </div>
              </span>
            </a>
          </li>
          <li>
            <a
              className={null}
              href="/changelog"
              onClick={[Function]}
            >
              Changelog
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <h1>
    Hello
  </h1>
  <footer
    className="emotion-9"
  >
    <div
      className="emotion-4"
    >
      <nav>
        <ul
          className="emotion-2 emotion-3"
        >
          <li>
            <a
              href="https://github.com/ricardocasares/analogical"
            >
              <svg
                height={12}
                viewBox="0 0 64 64"
                width={12}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M32 0 C14 0 0 14 0 32 0 53 19 62 22 62 24 62 24 61 24 60 L24 55 C17 57 14 53 13 50 13 50 13 49 11 47 10 46 6 44 10 44 13 44 15 48 15 48 18 52 22 51 24 50 24 48 26 46 26 46 18 45 12 42 12 31 12 27 13 24 15 22 15 22 13 18 15 13 15 13 20 13 24 17 27 15 37 15 40 17 44 13 49 13 49 13 51 20 49 22 49 22 51 24 52 27 52 31 52 42 45 45 38 46 39 47 40 49 40 52 L40 60 C40 61 40 62 42 62 45 62 64 53 64 32 64 14 50 0 32 0 Z"
                  fill="currentcolor"
                  strokeWidth="0"
                />
              </svg>
               Github
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/ricardocasares"
            >
              <svg
                height={12}
                viewBox="0 0 64 64"
                width={12}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M60 16 L54 17 L58 12 L51 14 C42 4 28 15 32 24 C16 24 8 12 8 12 C8 12 2 21 12 28 L6 26 C6 32 10 36 17 38 L10 38 C14 46 21 46 21 46 C21 46 15 51 4 51 C37 67 57 37 54 21 Z"
                  fill="currentcolor"
                  strokeWidth="0"
                />
              </svg>
               Twitter
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </footer>
</div>
`;
